### PR TITLE
Typechecker minor fixes

### PR DESCRIPTION
**Pull request by sornas**

### CODE REVIEW

File: sylt-compiler/src/typechecker.rs [hunk 1]
```diff
@@ -631,7 +625,10 @@ impl TypeChecker {
             | StatementKind::Blob { .. }
             | StatementKind::IsCheck { .. }
             | StatementKind::ExternalDefinition { .. } => {
-                unreachable!("Illegal inner statement! Parser should have caught this.")
+                unreachable!(
+                    "Illegal inner statement at {:?}! Parser should have caught this.",
+                    span
```
Comment (FredTheDino) [inline sylt-compiler/src/typechecker.rs:630]: Isnt this debug print is really bad? Maybe say which line and rsnge it has instead?

File: sylt-compiler/src/typechecker.rs [hunk 2]
```diff
@@ -1499,24 +1486,17 @@ impl TypeChecker {
...
-                        )
-                        .help(
-                            self,
-                            var.span,
-                            "Originally defined here".into(),
-                        )
-                    }
-                } else {
-                    match self.globals.get(&(ctx.namespace, ident.name.clone())) {
+                match self.stack.iter().rfind(|v| v.ident.name == ident.name) {
```
Comment (FredTheDino) [inline sylt-compiler/src/typechecker.rs:1489]: Nice!

**Review by FredTheDino — APPROVED** [review]
Lgtm, nothing controversial! Maybe we should add a better print for spans?

**Review by EMattfolk — APPROVED** [review]

File: sylt-compiler/src/typechecker.rs (hunk 1 above)
Comment (sornas) [reply sylt-compiler/src/typechecker.rs:630]: kinda bad, yes, but i don't know how to get the filename from just the span.

Comment (FredTheDino) [reply sylt-compiler/src/typechecker.rs:630]: You can check the type_error macro - the span has a `file_id` that indexes `namespace_to_filename` (I think these names are right, I might be wrong.)

**Comment by FredTheDino** [comment]
This code is getting outdated - I'd recomend to merge it ASAP <3

File: sylt-compiler/src/typechecker.rs [hunk 3]
```diff
@@ -655,7 +654,12 @@ impl TypeChecker {
             | StatementKind::StatementExpression { .. }
             | StatementKind::Unreachable
             | StatementKind::EmptyStatement => {
-                panic!("Illegal outer statement! Parser should have caught this")
+                panic!(
```
Comment (EMattfolk) [inline sylt-compiler/src/typechecker.rs:657]:
```suggestion
                unreachable!(
```
This is supposed to unreachable right?

**Review by EMattfolk — APPROVED** [review]

File: sylt-compiler/src/typechecker.rs (hunk 3 above)
Comment (sornas) [reply sylt-compiler/src/typechecker.rs:657]: yes :+1:

**Pull request merged**